### DISCUSSION TITLE
How does Fly Around works?

### DISCUSSION THREAD

**Discussion by AleksandarIlic** [NVlabs/instant-ngp#284] [Q&A] [answered]
Hi all,

I was wondering how the fly around a scene (e.g. fly around factory video) works and I have some very basic questions, hopefully someone can help me out with them:

1. Is the fly around a scene an output of the learning and training process, where you have to program the camera path and etc before starting the process?

2. Or fly around a scene can be programmed or recorded as you wish after the learning and training process has completed?

Thanks,
Alex

**Comment by Tom94** [comment, marked as answer]
Hi Alex, for the `.gif`s at the top of the readme, I recorded my screen while interactively exploring the models.

If you want to create high-quality videos like those from the project page, you can create a camera path using the GUI (there's a widget just for camera path editing) and then play it back + render it through the Python bindings. The latter allow for all kinds of flexibility: you can interleave the playback of the camera path with training, load a previously trained snapshot, or do something else that you come up with.

Cheers!

**Reply by felixgeen** [reply]
Hi Tom,

Thanks for all your helpful replies I've been reading throughout the discussions. I wanted to ask further about how I might "interleave the playback of the camera path with training" I have been able to successfully load the pre-trained data and render out a high quality video with that. But I would love to be able to have a camera path move along while also showing the "training process" happening at the same time. How would I achieve this? I have been using byclouds script here: https://github.com/bycloudai/instant-ngp-Windows/blob/master/scripts/render.py

Thank you.

**Comment by AleksandarIlic** [comment]
Hi Thomas, thanks for the answer and congratulations to the amazing work you guys have done so far.

Since training is the most expensive operation I was hoping that it would be possible to do the training once and then render videos afterwards. I'm glad that this is supported!

Thanks again!
